### PR TITLE
Fixes for Moodle 3.11 Release

### PR DESCRIPTION
**Pull request by sbourget**
Includes the following functionality:
1. Fix for issue #72
2. Migration from Travis-CI to Github actions
3. Assorted coding style fixes to make Moodle codechecker  happy
4. Conversion to use new Activity Completion APIs
5. Includes version bump to version 3.11-r1 (And dependency of Moodle 3.11)

### CODE REVIEW

File: amd/src/quizgame.js [hunk 1]
```diff
@@ -340,10 +392,18 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
         return questions[level].question;
     }
 
+    /**
+     * Helper funcction to place text on screen
```
Comment (xow) [inline amd/src/quizgame.js:396]:
```suggestion
     * Helper function to place text on screen
```

File: amd/src/quizgame.js [hunk 2]
```diff
@@ -373,8 +433,15 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
         }
     }
 
+    /**
+     * Helper function ?
```
Comment (xow) [inline amd/src/quizgame.js:437]:
```suggestion
     * Helper function main game logic: process movements and behaviours of game objects
```

File: amd/src/quizgame.js [hunk 3]
```diff
@@ -396,33 +463,49 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
         }
     }
 
-    function Rectangle(left, top, width, height)
-    {
+    /**
+     * Helper Function ??
```
Comment (xow) [inline amd/src/quizgame.js:467]:
```suggestion
     * Old style JS constructor for storing information about a rectangle shape
```

This does need to be replaced with a proper class definition, assuming moodle supports es6 these days

File: amd/src/quizgame.js [hunk 4]
```diff
@@ -435,40 +518,52 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
...
         return new Rectangle(this.x, this.y, this.image.width, this.image.height);
     };
-    GameObject.prototype.die = function () {
+
+    GameObject.prototype.die = function() {
         this.alive = false;
     };
 
+    /**
+     * Helper Function ??
```
Comment (xow) [inline amd/src/quizgame.js:552]:
```suggestion
     * Constructor for Player class, all the information about the player
```

File: amd/src/quizgame.js [hunk 5]
```diff
@@ -521,16 +618,31 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
         }
     };
 
+    /**
+     * Helper Function for background objects
```
Comment (xow) [inline amd/src/quizgame.js:622]:
```suggestion
     * Constructor for Planet (background objects) extends GameObject
```

File: amd/src/quizgame.js [hunk 6]
```diff
@@ -521,16 +618,31 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
...
     Planet.prototype = Object.create(GameObject.prototype);
-    Planet.prototype.update = function (bounds) {
+    Planet.prototype.update = function(bounds) {
         planet.image.width = displayRect.width;
         planet.image.height = displayRect.height;
         GameObject.prototype.update.call(this, bounds);
     };
 
+    /**
+     * Helper Function for enemy craft (answers)
```
Comment (xow) [inline amd/src/quizgame.js:639]:
```suggestion
     * Constructor for enemy craft (answers) extends GameObject
```

File: amd/src/quizgame.js [hunk 7]
```diff
@@ -629,10 +749,19 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
         currentTeam = [];
     }
 
+    /**
+     * Helper function for T/F questions
```
Comment (xow) [inline amd/src/quizgame.js:753]:
```suggestion
     * Helper function for True/False questions
```

File: amd/src/quizgame.js [hunk 8]
```diff
@@ -654,11 +784,21 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
         }
     };
 
+    /**
+     * Helper function for MCQ questions
```
Comment (xow) [inline amd/src/quizgame.js:788]:
```suggestion
     * Helper function for multiple choice questions (MCQ)
```

File: amd/src/quizgame.js [hunk 9]
```diff
@@ -679,6 +820,15 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
...
```
Comment (xow) [inline amd/src/quizgame.js:824]:
```suggestion
     * Helper function for matching questions
```

File: amd/src/quizgame.js [hunk 10]
```diff
@@ -753,14 +908,22 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
         this.hightlighted = false;
     };
 
+    /**
+     * Helper function for laser shots
```
Comment (xow) [inline amd/src/quizgame.js:912]:
```suggestion
     * Constructor for laser shots extends GameObject
```

File: amd/src/quizgame.js [hunk 11]
```diff
@@ -770,13 +933,21 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
...
+
+    Laser.prototype.deflect = function() {
         this.image = this.loadImage("pix/enemylaser.png");
         this.direction.y *= -1;
         this.friendly = !this.friendly;
         playSound("deflect");
     };
 
+    /**
+     * Helper function for background
```
Comment (xow) [inline amd/src/quizgame.js:945]:
```suggestion
     * Constructor for explosion particle effects extends GameObject
```

File: amd/src/quizgame.js [hunk 12]
```diff
@@ -801,15 +974,21 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
...
-    Particle.prototype.draw = function (context) {
+
+    Particle.prototype.draw = function(context) {
         context.fillStyle = this.colour;
         context.fillRect(this.x, this.y, this.width, this.height);
         context.stroke();
     };
 
+    /**
+     * Helper function for background stars
```
Comment (xow) [inline amd/src/quizgame.js:989]:
```suggestion
     * Helper function for background stars extends GameObject
```

File: amd/src/quizgame.js [hunk 13]
```diff
@@ -819,23 +998,37 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
...
-    Star.prototype.draw = function (context) {
+
+    Star.prototype.draw = function(context) {
         context.fillStyle = '#9999AA';
         context.fillRect(this.x, this.y, this.width, this.height);
         context.stroke();
     };
 
+    /**
+     * Helper funcction to handle collisions.
```
Comment (xow) [inline amd/src/quizgame.js:1016]:
```suggestion
     * Helper function to handle collisions between gameobjects.
```

File: amd/src/quizgame.js [hunk 14]
```diff
@@ -855,20 +1048,44 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
                 return true;
             }
         }
+        return false;
     }
 
+    /**
+     * Helper funcction to handle collisions.
```
Comment (xow) [inline amd/src/quizgame.js:1055]:
```suggestion
     * Helper function to handle intersections between GameObjects.
```

File: amd/src/quizgame.js [hunk 15]
```diff
@@ -855,20 +1048,44 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
...
+     * @return {boolean}
+     */
     function objectsIntersect(object1, object2) {
         var rect1 = object1.getRect();
         var rect2 = object2.getRect();
         return rect1.Intersect(rect2);
     }
 
+    /**
+     * Helper function for background spray
```
Comment (xow) [inline amd/src/quizgame.js:1067]:
```suggestion
     * Helper function for spraying particle (explosion) effects
```

File: amd/src/quizgame.js [hunk 16]
```diff
@@ -988,25 +1234,41 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
         }
     }
 
+    /**
+     * Helper function for mouse movement.
```
Comment (xow) [inline amd/src/quizgame.js:1238]:
```suggestion
     * Helper function for mouse release. (stops the Player) for mouse mode
```

File: amd/src/quizgame.js [hunk 17]
```diff
@@ -974,6 +1216,10 @@ define(['jquery','core/yui', 'core/notification', 'core/ajax'], function($, Y, n
...
```
Comment (xow) [inline amd/src/quizgame.js:1220]:
```suggestion
     * Helper function for mouse click (starts the player shooting if you clicked on them, moving if you didn't).
```

**Review by xow — APPROVED** [review]
Looks good, I made some suggestion tweaks. You can apply them by going "add to batch" and commit them all at once. Haven't run it yet as my VPS is out of hard disk space... working on it

**Comment by sbourget** [comment]
I've merged in your changes and made a few additional fixes (as moodle core had made some breaking changes in the last few weeks of the beta.  I re-ran grunt and both behat and PHPunit are passing:

https://github.com/sbourget/moodle-mod_quizgame/runs/2632304525?check_suite_focus=true

Let me know if you want to test it locally or if I can merge and then tag the new release.

**Comment by xow** [comment]
Merging now as I couldn't get the latest moodle working, I'll have to upgrade my setup another time

**Pull request merged**